openapi3/schema.go

```diff
@@ -1222,6 +1223,19 @@ func (err *SchemaError) Error() string {
 func isSliceOfUniqueItems(xs []interface{}) bool {
 	s := len(xs)
 	m := make(map[interface{}]struct{}, s)
+	xsKind := reflect.TypeOf(xs[0]).Kind()
+	if xsKind == reflect.Map ||
+		xsKind == reflect.Slice {
+		for i := 0; i < s; i++ {
+			m[fmt.Sprintf("%v", xs[i])] = struct{}{}
```

Comment:
This can collide with strings. How about making a `map[string]struct{}` and creating keys with `json.Marshal`? This allows dropping `reflect` too.

Reply:
The reason why I use reflect to distinguish Map and slice and also choose fmt.Sprintf is due to I do some benchmark test, result as below:
When input array item is type string or number, the array legnth is 10:
```
BenchmarkIsSliceOfUniqueItemsOnlyMapSliceUseJSONMarshal-8   	 2503779	       475 ns/op	     291 B/op	       1 allocs/op
BenchmarkIsSliceOfUniqueItemsOnlyMapSliceUseSprintf-8       	 2524917	       484 ns/op	     291 B/op	       1 allocs/op
BenchmarkIsSliceOfUniqueItemsAllJSONMarshal-8               	  442516	      2719 ns/op	     387 B/op	      12 allocs/op
BenchmarkIsSliceOfUniqueItemsAllSprintf-8                   	 1000000	      1104 ns/op	     291 B/op	       1 allocs/op
```
When input array item is type array,outter array length is 10,  inner array is 2:
```
BenchmarkIsSliceOfUniqueItemsOnlyMapSliceJSONMarshal-8   	  225642	      5381 ns/op	    1091 B/op	      31 allocs/op
BenchmarkIsSliceOfUniqueItemsOnlyMapSliceSprintf-8       	  254638	      4682 ns/op	     771 B/op	      21 allocs/op
BenchmarkIsSliceOfUniqueItemsAllJSONMarshal-8            	  249345	      4922 ns/op	     947 B/op	      22 allocs/op
BenchmarkIsSliceOfUniqueItemsAllSprintf-8                	  298896	      4233 ns/op	     611 B/op	      11 allocs/op
```

Acctualy, I would like suggest we can let user register their own function used to check unique of array, which might have better performance when that function is designed for specific input data.

Reply:
Agreed & I think this is a very good idea. Can you provide a default registered function that uses json.Marshal? Thanks

Reply:
New codes provided.